update analysis README.md (#102)
redhat-cop/infra.leapp: roles/analysis/README.md

# Analysis

The `analysis` role is used to create the Leapp pre-upgrade report on the target hosts.

This is used in IPU planning to identify unhandled cases, that would result in process failure.

This role also saves a copy of the current ansible facts under the `/etc/ansible/facts.d` directory for validation after upgrade.

## Requirements

Metadata is required for leap preupgrade analysis to run.  When a system uses CDN, this is provided automatically.  For systems not on CDN, package metadata files needs to be provided by URL in leapp_metadata_url. Once obtained during IPU, file is unpacked on IPU host under `/etc/leapp/files` directory.

In an upcomoing release, the metadata will be included as part of the RPM install of leapp RPMs.

For more information, refer to the knowledge article [Leapp utility metadata in-place upgrades of RHEL for disconnected upgrades (including Satellite)](https://access.redhat.com/articles/3664871).

## Commonly modified variables

| Name                  | Type | Default value           | Description                                     |
|-----------------------|------|-------------------------|-------------------------------------------------|

> | leapp_upgrade_type    | String  | satellite | satellite, cdn or rhui |

Reviewer: Previous description was better. Restore like this... 

```suggestion
| leapp_upgrade_type    | String  | "satellite" | Set to "cdn" for hosts registered with Red Hat CDN and "rhui" for hosts using rhui repos. |
```

> | leapp_preupg_opt | String | --no-rhsm when leapp_upgrade_type is connected | Upstream repository usage - whether to use RHSM |

Reviewer: Previous description was better. Explanation of the default value is confusing. If we really want to provide a way for users to append extra options to the leapp command, we should use a different variable that won't interfere with our default logic. This is an optional variable, so it should go in your new optional section. 

Move to optional section and change to this...
```suggestion
| leapp_preupg_opt | String |  | Optional string to define command line options to be passed to the `leapp` command when running the pre-upgrade. |
```

> | leapp_metadata_url      | String |  | URL to the leapp metadata, usually over https |

Reviewer: Better to reference the detailed explanation rather than the same thing explained differently in two places. 

```suggestion
| leapp_metadata_url      | String |  | See Requirements section above. |
```

Author: Removed Metadata references

> | leapp_enable_repos_args | String |  | --enablerepo (leapp_repos_enabled) or blank |

Reviewer: Remove this. Role users should not be messing with it. 

```suggestion
```

| leapp_preupg_opt | String | --no-rhsm when leapp_upgrade_type is connected | Upstream repository usage - whether to use RHSM |
| leapp_metadata_url      | String |  | URL to the leapp metadata, usually over https |
| leapp_enable_repos_args | String |  | --enablerepo (leapp_repos_enabled) or blank |

## Role variables

| Name                    | Default value         | Description                                         |
|-------------------------|-----------------------|-----------------------------------------------------|
| leapp_upgrade_type      | "satellite"           | Set to "cdn" for hosts registered with Red Hat CDN and "rhui" for hosts using rhui repos. |
| leapp_metadata_url      |                       | See Requirements section above.                     |
| leapp_answerfile        |                       | Optional multi-line string. If defined, this will be used as the contents of `/var/log/leapp/answerfile`. |
| leapp_preupg_opts       |                       | Optional string to define command line options to be passed to the `leapp` command when running the pre-upgrade. |

> | post_reboot_delay       | 120                   | Optional integer to pass to the reboot post_reboot_delay option. |

Reviewer: Why did we remove post_reboot_delay variable. Add it back under optional section.

> | satellite_organization  | String | Example | Organization used in Satellite definition |

Reviewer: "Example" is not the default value. That line in defaults/main.yml is commented out. 

```suggestion
| satellite_organization  | String |  | Organization used in Satellite definition |
```

> | satellite_activation_key_leapp | String |  | Key used to identify activation key |

Reviewer: Change like this... 

```suggestion
| satellite_activation_key_leapp | String |  | Satellite activation key |
```

> | satellite_activation_key_leapp     | String |  | Post-IPU state of subscriptions and svc level |

Reviewer: I have no idea what this means. If role users really need to use these variables, we need to explain them much better than a short description that will fit in this table. Also, s/IPU/upgrade/g.

> | leapp_repos_enabled    | List | [] | Satellite repo for the satellite client RPM install |

Reviewer: Same question as above. If this needs to be set in some specific way for Satellite users, it needs to be explained in more detail.

Author: removed the Post-activation key and added section for the initial activation key and the leapp activation key,

| Name                  | Type | Default value           | Description                                     |
|-----------------------|------|-------------------------|-------------------------------------------------|

> | result_filename | String |  | Path to file for the output of leapp |
> | analysis_packages_el6 | List | redhat-upgrade-tool | RPMS that need to be installed for IPU to RHEL7 |
> | analysis_repos_el6    | List | rhel-6-server-extras-rpms rhel-6-server-optional-rpms | Repo to be enabled for IPU to RHEL7  |
> | analysis_packages_el7 | List | leapp-upgrade             | RPMS that need to be installed for IPU to RHEL7 |
> | analysis_repos_el7    | List | rhel-7-server-extras-rpms | Repo to be enabled for IPU to RHEL7 |
> | analysis_packages_el8 | List | leapp-upgrade | RPMS that need to be installed for IPU to RHEL8 |
> | analysis_repos_el8 | List | rhel-7-server-extras-rpms | Repo to be enabled for IPU to RHEL7             |

Reviewer: Remove these from the doc. Should not be used by role users... 

```suggestion
```

Author: removd.

> | leapp_answerfile | TXT file | /var/log/leapp/answerfile | Optional - Source for Alternate AnswerFile needed during leapp process while upgrading  |

Reviewer: No, this is wrong. The variable has no default value and it doesn't define the filename of the answerfile. It is used to define the contents of the answerfile. Please fix as follows:

```suggestion
| leapp_answerfile | Multi-line String |  | If defined, this writen to `/var/log/leapp/answerfile` before generating the pre-upgrade report. |
```

Author: ok
